### DISCUSSION TITLE
Doirs 的 INTERSECT  关键字实现逻辑

### DISCUSSION THREAD

**Discussion by qiaoshiling** [apache/doris#23918] [A - General / Q&A]
Doris版本 1.2.5
我现在有一张表a，想取到他在三种不同条件下的交集 使用到了INTERSECT关键字
测试语句

```plpgsql

 (
SELECT
	a.sku_code
FROM
	(
	   SELECT  *  FROM	a
    ) a
WHERE
	  a.is_lst > 0 AND a.site_code = 'UK'
 )
INTERSECT
 (
SELECT
     b.sku_code
FROM
    (
       SELECT * FROM a 
    ) b
WHERE
      b.is_lst >= 0 AND b.site_code = 'DE'
 )
INTERSECT
 (
SELECT
     c.sku_code
FROM
    (
     SELECT  * FROM  a 
    ) c
 WHERE
     c.is_lst >= 0   AND c.site_code = 'ES'
 )
``` 

每次查询结果数据量从 198~200 不等
后经过sql 修改
```plpgsql
select
	mqxwmfam.*
from
	((
	SELECT
		a.sku_code
	FROM
	    (
		select * from	a
         ) a
where
   a.is_lst > 0 AND a.site_code = 'UK')
INTERSECT
		(
	SELECT
		b.sku_code
	FROM
		(
		select * from	a
		) b
	WHERE
		b.is_lst >= 0 AND b.site_code = 'DE' ) ) mqxwmfam
INTERSECT
	(
    SELECT
		c.sku_code
    FROM
		(
	select	* from	a
	    ) c
WHERE
 c.is_lst >= 0	AND c.site_code = 'ES' )
```

可以看到我将第一个 INTERSECT 变为了一个子查询并取了别名再进行交集运算，数据稳定在了201条。
是否可以告知我Doris INTERSECT关键字的实现逻辑以及为什么会出现这样的问题
如果更多的子集求交集应该怎么写 一层一层的做嵌套么？

**Comment by yagagagaga** [comment]
Could you provide the tables DDL?

**Reply by qiaoshiling** [reply]
This is my table ddl:
CREATE TABLE `ads_sku_site_basic_label_d` (
  `stat_day` varchar(200) NOT NULL COMMENT '',
  `stat_date` datetime NULL COMMENT '',
  `sku_code` varchar(200) NOT NULL COMMENT '',
  `sku_name` varchar(200) NULL COMMENT '',
  `site_code` varchar(200) NOT NULL COMMENT '',
  `group_code` varchar(200) NULL COMMENT '',
  `group_name` varchar(200) NULL COMMENT '',
  `fst_lst_time` datetime NULL COMMENT '',
  `lst_days` bigint(20) NULL COMMENT '',
  `promt_price` decimal(20, 2) NULL COMMENT '',
  `promt_profit_rate` decimal(20, 2) NULL COMMENT '',
  `feed_title` varchar(1000) NULL COMMENT '',
  `lifecycle_name` varchar(200) NULL COMMENT '',
  `is_lst` int(11) NULL COMMENT '',
  `is_bulk_prd` int(11) NULL COMMENT '',
  `is_must_buy` int(11) NULL COMMENT '',
  `revw_cnt` bigint(20) NULL COMMENT '',
  `avg_rating` decimal(20, 2) NULL COMMENT '',
  `wos` decimal(20, 2) NULL COMMENT '',
  `sales_inv_qty` bigint(20) NULL COMMENT '',
  `on_trans_mid_inv_qty` bigint(20) NULL COMMENT '',
  `on_trans_last_inv_qty` bigint(20) NULL COMMENT '',
  `se_exp_cnt_1d` bigint(20) NULL COMMENT '',
  `a2w_clk_pv_1d` bigint(20) NULL COMMENT '',
  `view_pbd_pv_1d` bigint(20) NULL COMMENT '',
  `view_pbd_uv_1d` bigint(20) NULL COMMENT '',
  `a2c_clk_uv_1d` bigint(20) NULL COMMENT '',
  `sales_tred_w` varchar(1000) NULL COMMENT '',
  `sales_tred_m` varchar(1000) NULL COMMENT '',
  `cvr_w_avg` decimal(20, 2) NULL COMMENT '',
  `dvat_4w` decimal(20, 2) NULL COMMENT '',
  `dvat_8w` decimal(20, 2) NULL COMMENT '',
  `dvat_12w` decimal(20, 2) NULL COMMENT '',
  `is_clr_wms` int(11) NULL COMMENT '',
  `dim_cube_code` varchar(200) NULL COMMENT ''
) ENGINE = OLAP DUPLICATE KEY(`stat_day`) COMMENT '' DISTRIBUTED BY HASH(`stat_day`, `sku_code`, `site_code`) BUCKETS 16 PROPERTIES (
  "replication_allocation" = "tag.location.default: 1",
  "in_memory" = "false",
  "storage_format" = "V2",
  "enable_unique_key_merge_on_write" = "true",
  "disable_auto_compaction" = "false"
);

**Comment by yagagagaga** [comment]
This is indeed a bug, you can upgrade to 2.0.1 to solve this problem.
